File: web-development/level-1/helsinki-part-1-2/Java/part1/week5/week05-ex84-OverloadedCounter/Counter.java
public class Counter  {

  //fields
  private int value;
  private boolean check;

  //constructors
  public Counter(int startingValue, boolean check)  {
    //creates a new counter with the given value. The check is on if the parameter given to check was true.
    this.value = startingValue;
    if (check == true)  {
      this.check = true;
    }
  }

  public Counter(int startingValue)  {
    //creates a new counter with the given value. The check on the new counter should be off.
    this.value = startingValue;
    this.check = false;
  }

  public Counter(boolean check) {
    //creates a new counter with the starting value 0. The check is on if the parameter given to check was true.
    this.value = 0;
    if (check == true)  {
      this.check = true;
    }
  }

  public Counter()  {
    //creates a new counter with the starting value of 0 and with checking off.
    this.value = 0;
    this.check = false;
  }

  //behaviors
  public int value()  {
    return this.value;
  }

  public void increase()  {
    this.value++;
  }

  public void decrease()  {
    //decreases the value but cannot go below 0 if check is on
    if (this.value == 0 && this.check == true)  {
      this.value = 0;
    }
    else if (this.value > 0)  {
      this.value--;
    }
  }



}

File: web-development/level-1/helsinki-part-1-2/Python/part1/week05/ex84/Counter.py
class Counter:

    #constructors
    def __init__(self, starting_value):
        self.value = starting_value

    #behaviors

    #gets the value
    def get_value(self):
        return self.value

    #increase the counter value by one
    def increase(self):
        self.value += 1

    #decreases the counter value by one but cannot go below 0
    def decrease(self):
        if self.value > 0:
            self.value -= 1
        elif self.value == 0:
            print('Cannot go below 0')
